crates/runtime/src/logic.rs

```diff
@@ -556,4 +566,24 @@ impl VMHostFunctions<'_> {
 
         Ok(())
     }
+
+    /// Call the contract's `send_proposal()` function through the bridge.
+    ///
+    /// The proposal actions are obtained as raw data and then deserialised into
+    /// the individual actions that make up the proposal.
+    ///
+    /// # Parameters
+    ///
+    /// * `ptr` - Pointer to the start of the action data in WASM memory.
+    /// * `len` - Length of the action data.
+    ///
+    pub fn send_proposal(&mut self, ptr: u64, len: u64) -> VMLogicResult<()> {
+        // TODO: This will need to deserialise into ProposalActions when available
+        // TODO: So this will become Vec<ProposalAction>
+        let actions: Vec<Vec<u8>> = from_borsh_slice(&self.read_guest_memory(ptr, len)?)
```

Comment:
the runtime shouldn't care about the type being returned here, just receive a `Vec<u8>` and push it to a list of proposals on `VMLogic`, like we did with `send_action`

Reply:
Ahhhh, okay - I think that makes sense. Changed!